snowdrop-block/src/Snowdrop/Block/Application.hs

```diff
@@ -52,7 +52,7 @@ instance Buildable blockRef => Buildable (BlockApplicationException blockRef) wh
 
 data ForkApplyAction chgAccum blkType = ForkApplyAction
     { faaRollbackTo  :: BlockRef blkType
-    , faaApplyBlocks :: OldestFirst [] (Block (BlockHeader blkType) (Payload blkType), chgAccum)
+    , faaApplyBlocks :: OldestFirst [] (Block (BlockHeader blkType) (Payload blkType), [chgAccum])
```

Comment:
Semantics behind previous version of this data type implied that along with block a change accumulator was returned as for the state with the block (and all preceding blocks in this `ForkApplyAction` applied).

Current signature would make sense if we wanted to have change accumulator along with each transaction, but surely this isn't needed.

Reply:
Yes, for every `tx` in a block we have its `chgAccum` as signature of `bscExpandPayloads` says:

```
OldestFirst [] (OldestFirst [] (Tx blkType, chgAccum))
```

`faaApplyBlocks` signature on the other hand says we have one `chgAccum` per block:

```
type Payload blkType = OldestFirst [] (Tx blkType)
OldestFirst [] (Block (BlockHeader blkType) (Payload blkType), chgAccum)
```

This make sense but that means we need a function from `[chgAccum]` to `chgAccum` but such a function doesn't exists, or does it? I'm missing something here...

Reply:
Such function is `head`

Reply:
How simple :)